src/Proto3/Suite/DotProto/Parsing.hs

```diff
@@ -378,10 +383,11 @@ pOptionKw = do
 --------------------------------------------------------------------------------
 -- service statements
 
-servicePart :: ProtoParser DotProtoServicePart
-servicePart = DotProtoServiceRPCMethod <$> rpc
-          <|> DotProtoServiceOption <$> pOptionStmt
-          <|> DotProtoServiceEmpty <$ empty
+servicePart :: ProtoParser (Maybe DotProtoServicePart)
+servicePart =
+  (fmap (Just . DotProtoServiceRPCMethod) rpc)
+    <|> (fmap (Just . DotProtoServiceOption) pOptionStmt)
```

Comment:
Presumably you used them for stylistic reasons, but the parentheses around the arguments to `<|>` are unnecessary.

Reply:
Yes, we probably do. If we don't need `try` here then adding it wouldn't hurt.

Reply:
Thanks.